src/Microsoft.ML.StaticPipe/StaticSchemaShape.cs

```diff
@@ -327,34 +327,22 @@ private static Type GetTypeOrNull(Schema.Column col)
         /// type for communicating text.
         /// </summary>
         /// <returns>The basic type used to represent an item type in the static pipeline</returns>
-        private static Type StaticKind(DataKind kind)
+        private static Type GetPhysicalType(ColumnType columnType)
         {
-            switch (kind)
+            switch (columnType)
             {
-                // The default kind is reserved for unknown types.
-                case default(DataKind): return null;
-                case DataKind.I1: return typeof(sbyte);
-                case DataKind.I2: return typeof(short);
-                case DataKind.I4: return typeof(int);
-                case DataKind.I8: return typeof(long);
-
-                case DataKind.U1: return typeof(byte);
-                case DataKind.U2: return typeof(ushort);
-                case DataKind.U4: return typeof(uint);
-                case DataKind.U8: return typeof(ulong);
-                case DataKind.U16: return typeof(RowId);
-
-                case DataKind.R4: return typeof(float);
-                case DataKind.R8: return typeof(double);
-                case DataKind.BL: return typeof(bool);
-
-                case DataKind.Text: return typeof(string);
-                case DataKind.TimeSpan: return typeof(TimeSpan);
-                case DataKind.DateTime: return typeof(DateTime);
-                case DataKind.DateTimeZone: return typeof(DateTimeOffset);
+                case NumberType numberType:
+                case KeyType keyType:
+                case TimeSpanType timeSpanType:
+                case DateTimeType dateTimeType:
+                case DateTimeOffsetType dateTimeOffsetType:
+                case BoolType boolType:
+                    return columnType.RawType;
+                case TextType textType:
+                    return typeof(string);
```

Comment:
FYI this style of change you have here is more what I'd be more comfortable with in the the `OnnxUtils` case. (Now, in that other case, it would also probably benefit from some `when` statements to further refine.)